### DISCUSSION TITLE
Questions about purging

### DISCUSSION THREAD

**Discussion by eduardonwa** [tailwindlabs/tailwindcss#3825] [Help] [answered]
By default when one does ```npx tailwindcss init``` the config file comes empty, I have 2 questions:

1. When you go into production, does this file "automatically" purges and clears all unused data? 
Seems like all the documentation talks about is _production_ and _updating_ in my case im not doing any of those.

2. Are there any commands to purge before production? should I just simply delete that file and re-run the command again? I know I could do that but figured there may be a command for that. 

3. If i were to purge today. what would happen to this:

```
@props([
    'hash',
    'type' => 'images',
    'height' => [
        'images' => 'h-64',
        'categories' => 'h-3/5'
    ]
])

        <div {{ $attributes->merge([ 
            'class' =>
                "{$height[$type]} fixed inset-0 flex items-center justify-center w-72 m-auto md:h-auto md:w-auto md:max-w-3xl md:my-5 md:mx-5 md:mx-auto bg-white rounded-md shadow-md overflow-x-auto md:overflow-hidden"
            ]) }}
        >
```
Thanks!

**Comment by simonswiss** [comment, marked as answer]
Hey!

1. Tailwind will not purge out of the box. The minimal amount of setup you need to do is provide an array of paths to all of your template files using the `purge` option:

https://tailwindcss.com/docs/optimizing-for-production#basic-usage

2. You can decide _when_ hour CSS will be purged. By default, it will only try purge when the `NODE_ENV` is set to `production`, but you can also control the logic for that inside of `purge.enabled` inside the config file:

https://tailwindcss.com/docs/optimizing-for-production#enabling-manually

3. Looks like all the class names in your code would be retained on purge, providing you provided the path to that file in your purge config. `{$height[$type]}` will also work because purge will be able to "see" the corresponding `h-64` it resolves to.

But if you had something like `<div class="h-{$height[$type]}"` (assuming this is the correct syntax), and your `height.images` prop was only `64`, then that would get purged out.

Purge needs to find **the entire class string** in the file, otherwise it's going to remove it from the CSS.

Hope it helps 👍
